src/rpm/package.rs

```diff
@@ -314,137 +314,124 @@ impl RPMPackageMetadata {
             .get_entry_data_as_string(IndexTag::RPMTAG_SOURCERPM)
     }
 
-    fn get_dependencies(
-        &self,
-        names_tag: IndexTag,
-        flags_tag: IndexTag,
-        versions_tag: IndexTag,
-    ) -> Vec<Dependency> {
-        // Return an empty list if the tags are not present
-        if self.header.find_entry_or_err(&names_tag).is_err() {
-            return vec![];
-        }
+    fn get_dependencies(&self, tags: &'static [IndexTag; 3]) -> Result<Vec<Dependency>, RPMError> {
+        let names = self.header.get_entry_data_as_string_array(tags[0]);
+        let flags = self.header.get_entry_data_as_u32_array(tags[1]);
+        let versions = self.header.get_entry_data_as_string_array(tags[2]);
```

Comment:
Why not just bail early here and use `map_err(Error::TagCoexistenceMismatch)` - if we need the error map at all.